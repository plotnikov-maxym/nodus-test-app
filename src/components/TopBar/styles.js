
export const styles = () => ({
  root: {
    width: '100%',
  },
  grow: {
    flexGrow: 1,
  },
});
