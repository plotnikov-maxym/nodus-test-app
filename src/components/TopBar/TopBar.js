import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import {styles} from './styles';
import {withStyles} from '@material-ui/core/styles';


const TopBar = ({classes, heading}) => {
  return (
      <div className={classes.root}>
        <AppBar position="static" color="primary">
          <Toolbar>
            <Typography variant="h5" color="inherit" noWrap>
              {heading}
            </Typography>
            <div className={classes.grow}/>
          </Toolbar>
        </AppBar>
      </div>
  );
};

TopBar.propTypes = {
  classes: PropTypes.object.isRequired,
  heading: PropTypes.string.isRequired,
};

export default withStyles(styles)(TopBar);
