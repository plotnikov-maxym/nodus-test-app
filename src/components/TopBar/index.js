import {connect} from "react-redux";
import TopBar from "./TopBar";

const mapStateToProps = state => ({
  isLoading: state.loading && state.loading.show,
});
export default connect(mapStateToProps)(TopBar);
