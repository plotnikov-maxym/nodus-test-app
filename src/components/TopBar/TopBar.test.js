import React from "react";
import renderer from "react-test-renderer";
import {extend} from "underscore";
import {TopBar} from "./TopBar";
import {createShallow} from "@material-ui/core/test-utils";
import Typography from "@material-ui/core/Typography";

// Mocks

describe("<TopBar />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  describe("TopBar component", () => {
    it("matches snapshot", async () => {
      const wrapper = await renderComponent();
      expect(wrapper).toMatchSnapshot();
    });

    it("renders the correct heading", async () => {
      const heading = "Howdy mates!";
      const wrapper = await getComponent({heading});
      expect(wrapper.find(Typography)).toExist();
      expect(wrapper.find(Typography).props().children).toEqual(heading);
    });
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<TopBar {...parsedProps} />);
  };

  const renderComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return renderer.create(<TopBar {...parsedProps} />).toJSON();
  };

  const getMockProps = () => ({
    classes: {
      root: "root",
      grow: "grow",
    },
    heading: "My Test Heading",
  });
});
