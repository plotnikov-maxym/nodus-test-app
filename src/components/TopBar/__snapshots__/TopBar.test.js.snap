// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> TopBar component matches snapshot 1`] = `
<div
  className="root"
>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8"
  >
    <div
      className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
      <h5
        className="MuiTypography-root-41 MuiTypography-h5-57 MuiTypography-colorInherit-70 MuiTypography-noWrap-67"
      >
        My Test Heading
      </h5>
      <div
        className="grow"
      />
    </div>
  </header>
</div>
`;
